src/linear_algebra/basic.lean

```diff
@@ -2417,32 +2417,32 @@ lemma neg_apply (x : M) : neg R x = -x := by simp
 
 end neg
 
-section ring
+section semiring
 
-variables [ring R] [add_comm_group M] [add_comm_group M₂]
+variables [semiring R] [add_comm_monoid M] [add_comm_monoid M₂]
 variables {module_M : module R M} {module_M₂ : module R M₂}
 variables (f : M →ₗ[R] M₂) (e : M ≃ₗ[R] M₂)
 
 /-- An `injective` linear map `f : M →ₗ[R] M₂` defines a linear equivalence
 between `M` and `f.range`. See also `linear_map.of_left_inverse`. -/
-noncomputable def of_injective (h : f.ker = ⊥) : M ≃ₗ[R] f.range :=
-of_left_inverse $ classical.some_spec (linear_map.ker_eq_bot.1 h).has_left_inverse
+noncomputable def of_injective (h : injective f) : M ≃ₗ[R] f.range :=
```

Comment:
I would imagine you can now move this right below `of_left_inverse` and reuse the existing section.

Reply:
done